app/models/dataset/model_builder.rb

```diff
@@ -0,0 +1,96 @@
+class Dataset
+  module Naming
+    def model_class_name(dataset_description)
+      "#{prefix}#{dataset_description.identifier}".classify
+    end
+
+    def full_model_class_name(dataset_description)
+      "Kernel::#{model_class_name(dataset_description)}"
+    end
+
+    def table_name(dataset_description)
+      prefix + dataset_description.identifier
+    end
+
+    def prefix
+      DatastoreManager.dataset_table_prefix
+    end
+
+    def association_name(dataset_description, morphed = false)
+      suffix = morphed ? "_morphed" : ""
+
+      :"#{prefix}#{dataset_description.identifier.pluralize}#{suffix}"
+    end
+  end
+
+  class ModelBuilder
+    include Naming
+
+    attr_reader :dataset_description
+
+    def initialize(dataset_description)
+      @dataset_description = dataset_description
+    end
+
+    def build
+      define_model_class
+      set_up_model
+      set_up_relation
+
+      model_class
+    end
+
+    def define_model_class
+      unless Kernel.const_defined?(model_class_name(dataset_description))
```

Comment:
@jsuchal ano. to bude dalsi krok, umoznosti redefinovanie. aktualne riesim to aby som oddelil casti do logickych celkov a odstranil previazanost. nasledne budem riesit to co pises. lebo chceme aby sa to dalo cachovat a nemuseli sa znova a znova definovat modely.